File: cPlusPlusCode/pencilSketchEdgeDetection.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace cv;
using namespace std;

int main(){

    //Read input image
    Mat image = imread("../assets/anish.jpg");

    //check if image exists
    if(image.empty()){
        cout<<"can not find image"<<endl;
        return 0;
    }

    //convert to gray scale
    Mat output;
    cvtColor(image, output, COLOR_BGR2GRAY);

    //Apply gaussian blur
    GaussianBlur(output, output, Size(3, 3), 0);

    //Detect edges
    Laplacian(output, output, -1, 5);

    //Invert the binary image
    output = 255 - output;

    //binary thresholding
    threshold(output, output, 150, 255, THRESH_BINARY);

    //create windows to display images
    namedWindow("image", WINDOW_AUTOSIZE);
    namedWindow("pencilsketch", WINDOW_AUTOSIZE);

    //display images
    imshow("image", image);
    imshow("pencilsketch", output);

    //Press esc to exit the program
    waitKey(0);

    //close all the opened windows
    destroyAllWindows();

    return 0;
}

File: pythonCode/pencilSketchEdgeDetection.py
import cv2
import sys

#read image
image = cv2.imread("../assets/anish.jpg")

#check if images exists
if image is None:
    print("can not find image")
    sys.exit()

#convert to gray scale
output = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)

#Apply gaussian blur
output = cv2.GaussianBlur(output, (3,3), 0)

#detect edges in the image
output = cv2.Laplacian(output, -1, ksize=5)

#invert the binary image
output = 255 - output

#binary thresholding
ret, output = cv2.threshold(output, 150, 255, cv2.THRESH_BINARY)

#create widnows to dispplay images
cv2.namedWindow("image", cv2.WINDOW_AUTOSIZE)
cv2.namedWindow("pencilsketch", cv2.WINDOW_AUTOSIZE)

#display images
cv2.imshow("image", image)
cv2.imshow("pencilsketch", output)

#press esc to exit the program
cv2.waitKey(0)

#close all the opened windows
cv2.destroyAllWindows()